buck2 fails to build on amd64

The package buck2 with version 20250301 and epoch 0 fails to build from source with the following error:

```
   Compiling derive_more v1.0.0
   Compiling semver v0.11.0
   Compiling pest_generator v2.7.15
   Compiling buck2_protoc_dev v0.1.0 (/home/build/app/buck2_protoc_dev)
   Compiling icu_locid_transform v1.5.0
   Compiling buck2_data v0.1.0 (/home/build/app/buck2_data)
   Compiling buck2_subscription_proto v0.1.0 (/home/build/app/buck2_subscription_proto)
   Compiling buck2_cli_proto v0.1.0 (/home/build/app/buck2_cli_proto)
   Compiling buck2_test_proto v0.1.0 (/home/build/app/buck2_test_proto)
   Compiling buck2_downward_api_proto v0.1.0 (/home/build/app/buck2_downward_api_proto)
   Compiling re_grpc_proto v0.1.0 (/home/build/remote_execution/oss/re_grpc_proto)
   Compiling buck2_action_metadata_proto v0.1.0 (/home/build/app/buck2_action_metadata_proto)
   Compiling buck2_forkserver_proto v0.1.0 (/home/build/app/buck2_forkserver_proto)
   Compiling buck2_worker_proto v0.1.0 (/home/build/app/buck2_worker_proto)
   Compiling buck2_install_proto v0.1.0 (/home/build/app/buck2_install_proto)
   Compiling pest_derive v2.7.15
   Compiling icu_properties v1.5.1
   Compiling starlark_syntax v0.13.0 (/home/build/starlark-rust/starlark_syntax)
   Compiling termwiz v0.18.0
   Compiling triomphe v0.1.14
   Compiling relative-path v1.9.3
   Compiling serde_bytes v0.11.17
   Compiling tracing-serde v0.2.0
   Compiling ipnetwork v0.20.0
   Compiling linked-hash-map v0.5.6
   Compiling erased-serde v0.3.31
   Compiling csv v1.3.1
   Compiling bincode v1.3.3
   Compiling sct v0.7.1
   Compiling rustls-webpki v0.101.7
   Compiling nibble_vec v0.1.0
   Compiling radix_trie v0.2.1
   Compiling dashmap v5.5.3
   Compiling parking_lot v0.11.2
   Compiling icu_normalizer v1.5.0
   Compiling crossterm v0.27.0
   Compiling futures-intrusive v0.4.2
   Compiling rustyline v14.0.0
   Compiling schemafy_core v0.5.2
   Compiling schemafy_lib v0.5.2
   Compiling tokio v1.44.1
   Compiling http v0.2.12
   Compiling serde_bser v0.4.0
   Compiling internment v0.8.6
   Compiling plist v1.7.0
   Compiling idna_adapter v1.2.0
   Compiling idna v1.0.3
error[E0283]: type annotations needed
   --> /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/plist-1.7.0/src/stream/binary_reader.rs:252:58
    |
252 |                 if value < 0 || value > u64::max_value().into() {
    |                                       -                  ^^^^
    |                                       |
    |                                       type must be known at this point
    |
    = note: multiple `impl`s satisfying `i128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
            - impl PartialOrd for i128;
            - impl<MIN, MAX> PartialOrd<deranged::RangedI128<MIN, MAX>> for i128
              where the constant `MIN` has type `i128`, the constant `MAX` has type `i128`;
help: try using a fully qualified path to specify the expected types
    |
252 |                 if value < 0 || value > <u64 as Into<T>>::into(u64::max_value()) {
    |                                         +++++++++++++++++++++++                ~

error[E0283]: type annotations needed
   --> /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/plist-1.7.0/src/stream/binary_reader.rs:252:58
    |
252 |                 if value < 0 || value > u64::max_value().into() {
    |                                                          ^^^^
    |
note: multiple `impl`s satisfying `_: From<u64>` found
   --> /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/plist-1.7.0/src/integer.rs:91:1
    |
91  | impl From<u64> for Integer {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/plist-1.7.0/src/value.rs:552:1
    |
552 | impl From<u64> for Value {
    | ^^^^^^^^^^^^^^^^^^^^^^^^
    = note: and more `impl`s found in the following crates: `core`:
            - impl From<u64> for AtomicU64;
            - impl From<u64> for i128;
            - impl From<u64> for u128;
    = note: required for `u64` to implement `Into<_>`
help: try using a fully qualified path to specify the expected types
    |
252 |                 if value < 0 || value > <u64 as Into<T>>::into(u64::max_value()) {
    |                                         +++++++++++++++++++++++                ~

   Compiling tracing-subscriber v0.3.19
   Compiling lsp-server v0.7.8
   Compiling schemafy v0.5.2
   Compiling crokey v1.1.0
   Compiling coolor v1.0.0
   Compiling url v2.5.4
For more information about this error, try `rustc --explain E0283`.
error: could not compile `plist` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
failed to build package: unable to run package buck2 pipeline: unable to run pipeline: exit status 101
```

--- a/buck2.yaml
+++ b/buck2.yaml
@@ -4,7 +4,7 @@ package:
   # version of rust that is compatible. To determine, go to the following URL:
   # - https://github.com/facebook/buck2/blob/<VERSION>/HACKING.md
   # Get the version, and update vars.rust-version below.
-  version: 20250301
+  version: 20250401
   epoch: 0
   description: "Build system, successor to Buck"
   copyright:
@@ -24,7 +24,7 @@ environment:
       - wolfi-base
 
 vars:
-  rust-version: nightly-2024-10-13
+  rust-version: nightly-2025-01-03
 
 var-transforms:
   - from: ${{package.version}}
@@ -36,7 +36,7 @@ pipeline:
   - uses: git-checkout
     with:
       repository: https://github.com/facebook/buck2
-      expected-commit: 060d0cf7b39eeefdafe38366b851fca9055e2fdd
+      expected-commit: 81c6b35d2a8e448e8283b8fe5fe7984d0c8ff31f
       tag: ${{vars.mangled-package-version}}
 
   - name: Set up Rust

--- a/buck2.yaml
+++ b/buck2.yaml
@@ -1,9 +1,5 @@
 package:
   name: buck2
-  # When bumping the version, you also need to retrieve the specific nightly
-  # version of rust that is compatible. To determine, go to the following URL:
-  # - https://github.com/facebook/buck2/blob/<VERSION>/HACKING.md
-  # Get the version, and update vars.rust-version below.
   version: 20250401
   epoch: 0
   description: "Build system, successor to Buck"
@@ -23,9 +19,6 @@ environment:
       - rustup
       - wolfi-base
 
-vars:
-  rust-version: nightly-2025-01-03
-
 var-transforms:
   - from: ${{package.version}}
     match: (\d{4})(\d{2})(\d{2})
@@ -41,15 +34,14 @@ pipeline:
 
   - name: Set up Rust
     runs: |
-      # This build requires a specific version of rust nightly.
-      # Be sure to check docs when bumping to more recent commits (see note at top of file).
-      rustup install ${{vars.rust-version}}
+      # This build requires a specific version of rust nightly: rustup will
+      # read upstream's rust-toolchain file to install the right one
+      rustup install
 
   - name: Configure and build
     runs: |
-      # Set up path
-      # This is a bit of a hack, but it's the easiest way to get the right version of rustc and cargo in the path.
-      export PATH="$HOME/.rustup/toolchains/${{vars.rust-version}}-${{host.triplet.rust}}/bin:$PATH"
+      # Include our nightly Rust bin/ directory in PATH
+      export PATH="$(dirname "$(rustup which cargo)"):$PATH"
 
       mkdir -p ${{targets.destdir}}/usr/bin/
       cd app/buck2
@@ -61,7 +53,6 @@ pipeline:
 
 update:
   enabled: true
-  manual: true
   ignore-regex-patterns:
     - 'latest'
   github:

